app/_hub/kong-inc/mocking/index.md

```diff
@@ -0,0 +1,897 @@
+---
+name: Mocking
+publisher: Kong Inc.
+version: 0.1.x
+# internal handler v 0.1.0
+
+desc: Provide mock endpoints to test your APIs against your services
+description: |
+  Provide mock endpoints to test your APIs in development against your services.
+  The Mocking plugin leverages standards based on the Open API Specification (OAS)
+  for sending out mock responses to APIs.
+
+  Benefits of service mocking with the Kong Mocking plugin:
+
+  - Conforms to a design-first approach since mock responses are within OAS.
+  - Accelerates development of services and APIs.
+  - Promotes parallel development of APIs across distributed teams.
+  - Provides an enhanced full lifecycle API development experience with Dev Portal
+    integration.
+  - Easily enable and disable the Mocking plugin for flexibility when
+    testing API behavior.
+
+enterprise: true
+type:
+  plugin
+categories:
+  - traffic-control
+
+kong_version_compatibility:
+    enterprise_edition:
+      compatible:
+        - 2.4.x
+params:
+  name: mocking
+  service_id: true
+  consumer_id: true
+  route_id: true
+  protocols: ["http", "https", "grpc", "grpcs"]
+  dbless_compatible: yes
+  dbless_explanation: |
+    Use the `api_specification` config for DB-less or hybrid mode. Attach the spec contents directly
+    instead of uploading to the Dev Portal. The API spec is configured directly in the plugin.
+  yaml_examples: false
+  k8s_examples: false
+  examples: false
+
+  config:
+    - name: api_specification_filename
+      required: semi
+      default:
+      datatype: string
+      value_in_examples:
+      description: |
+        The path and name of the specification file loaded into Kong DB. You cannot
+        use this option for DB-less or hybrid mode.
```

Comment:
It should be mutually exclusive but it isn't. You can use both. There is still the matter of precedence, which is unknown.

Reply:
@eskibars 

Sorry for getting to the party so late; would it be better to remove the `api_specification_filename` from the schema all together? This would avoid confusion by the user.

Reply:
> @eskibars
> 
> Sorry for getting to the party so late; would it be better to remove the `api_specification_filename` from the schema all together? This would avoid confusion by the user.

@mikefero that is what @murillopaula thought should happen, also making the  `api_specification` field required too.

Reply:
Chatted with @mikefero and I'll replay our conversation here.

I think we should leave this option in for the time being, though we should consider it for removal in 3.0.  We're sort of in a weird place with this plugin where even though it's been officially unreleased, it's kind of been used by a variety of folks and we should probably consider removal of this as a "breaking change in a minor" as strange as that may seem.  I agree it's a trappy configuration parameter and the non-file-based parameter should be strongly preferred (and encouraged).  Let's leave it in as-is for now, deprecate it in 2.5, and remove it in 3.0.